doc/source/generation.rst

```diff
@@ -1,5 +1,153 @@
+
+>>> g = Graph.Lattice(dim=[3, 3], circular=False)
+
+creates a 3x3 grid in two dimensions (9 vertices total). `circular` is used to connect each edge of the lattice back onto the other side, a process also known as "periodic boundary condition" that is sometimes helpful to smoothen out edge effects.
+
+The one dimensional, circular case (a ring) is important enough to deserve its own constructor :meth:`Graph.Ring`.
+
+.. note:: The line graph is a one dimensional, non-circular lattice, and can be also constructed as a non-circular ring.
+
+Graph atlas
++++++++++++
+The book ‘An Atlas of Graphs’ by Roland C. Read and Robin J. Wilson contains all undirected graphs with up to seven vertices, numbered from 0 up to 1252. You can create any graph from this list by index with :meth:`Graph.Atlas`, e.g.:
+
+>>> g = Graph.Atlas(44)
+
+The graphs are listed:
+
+ - in increasing order of number of nodes;
+ - for a fixed number of nodes, in increasing order of the number of edges;
+ - for fixed numbers of nodes and edges, in increasing order of the degree sequence, for example 111223 < 112222;
+ - for fixed degree sequence, in increasing number of automorphisms.
+
+
+Famous graphs
++++++++++++++
+A curated list of famous graphs, which are often used in the literature for benchmarking and other purposes, is available on the `igraph C core manual <https://igraph.org/c/doc/igraph-Generators.html#igraph_famous>`_. You can generate any graph in that list by name, e.g.:
+
+>>> g = Graph.Famous('Zachary')
+
+will teach you some about martial arts.
+
+
+Random graphs
++++++++++++++
+Stochastic graphs can be created according to several different models or games:
+
+ - Barabasi-Albert model: :meth:`Graph.Barabasi`
+ - Erdos-Renyi: :meth:`Graph.Erdos_Renyi`
+ - Watts-Strogatz :meth:`Graph.Watts_Strogatz`
+ - stochastic block model :meth:`Graph.SBM`
+ - forest fire game :meth:`Graph.Forest_Fire`
+ - random geometric graph :meth:`Graph.GRG`
+ - growing :meth:`Graph.Growing_Random`
+ - establishment game :meth:`Graph.Establishment`
+ - preference, the non-growing variant of establishment :meth:`Graph.Preference`
+ - asymmetric preference :meth:`Graph.Asymmetric_Prefernce`
+ - recent degree :meth:`Graph.Recent_Degree`
+ - K-regular (each node has degree K) :meth:`Graph.K_Regular`
+ - non-growing graph with edge probabilities proportional to node fitnesses :meth:`Graph.Static_Fitness`
+ - non-growing graph with prescribed power-law degree distributions :meth:`Graph.Static_Power_Law`
+ - in agreement with a sequence of degrees :meth:`Graph.Degree_Sequence`
+ - bipartite :meth:`Graph.Random_Bipartite`
+
+
+Other graphs
+++++++++++++
+Finally, there are some ways of generating graphs that are not covered by the previous sections:
+
+ - Kautz graphs :meth:`Graph.Kautz`
+ - De Bruijn graphs :meth:`Graph.De_Bruijn`
+ - Lederberg-Coxeter-Frucht graphs :meth:`Graph.LCF`
+ - graphs with a specified isomorphy class :meth:`Graph.Isoclass`
```

Comment:
"isomorphism", not "isomorphy" (apparently that's biology, not math)

Reply:
That's what I expected as well, but that's what the docstring says:

```python
Docstring:
Isoclass(n, class, directed=False)

Generates a graph with a given isomorphy class.

@param n: the number of vertices in the graph (3 or 4)
@param class: the isomorphy class
@param directed: whether the graph should be directed.
Type:      builtin_function_or_method
```

shall I change the docstring and docs both to `isomorphism`?

Reply:
Yes, I would be in favour of changing it to "isomorphism" everywhere in the docs.

However, this morning I did a more extensive search, and "isomorphy" does appear to be used in some mathematical papers. But it is exceedingly uncommon.

https://www.google.com/search?q=graph+%22isomorphy%22

Also, igraph comes up 2nd for in when I search for this term. 

"isomorphism" vs "isomorphy" 3.3 million vs 12,700 hits on Google, and 202,000 vs 1800 on Google Scholar.

Also: https://math.stackexchange.com/questions/982732/isomorphy-in-mathematical-texts

My conclusion: it is not incorrect (and it will certainly be understood), but it will sound strange to most contemporary mathematicians.

Reply:
yep agreed, done now.